### PR TITLE
feature/CORE-367/fix-message-block-in-resource-mode

### PR DESCRIPTION
**Pull request by bulatgab**
Fix for: the message block is not showing SMS/USSD/IVR content editors in the 'Resource mode'

Related PR: https://bitbucket.org/voto/voto5/pull-requests/3208

### CODE REVIEW

File: src/components/interaction-designer/resource-viewer/ResourceViewerBlock.vue [hunk 1]
```diff
@@ -1,8 +1,12 @@
...
   <div class="resource-viewer-block border p-2">
     <div class="block-header-1 d-flex">
-      <div class="mr-auto text-primary"><h5>{{blockLabel}}</h5></div>
-      <div class="ml-auto"><h6>{{trans(`flow-builder.${block.type}`).toUpperCase()}}</h6></div>
+      <div class="mr-auto text-primary">
+        <h5>{{ blockLabel }}</h5>
+      </div>
+      <div class="ml-auto">
+        <h6>{{ trans(`flow-builder.${block.type}`).toUpperCase() }}</h6>
+      </div>
```
Comment (bulatgab) [inline src/components/interaction-designer/resource-viewer/ResourceViewerBlock.vue:9]: fix ESLint issues

File: src/components/interaction-designer/resource-viewer/ResourceViewerBlock.vue [hunk 2]
```diff
@@ -11,10 +15,16 @@
...
+            v-if="!hasContent"
+            class="col-lg-3 col-sm-6 col-xs-12 pl-1 mb-3">
+            <div class="badge badge-info">
+              {{ trans('flow-builder.no-content-block-info') }}
+            </div>
           </div>
-          <slot name="tags" :block="block" />
+          <slot
+            name="tags"
+            :block="block" />
```
Comment (bulatgab) [inline src/components/interaction-designer/resource-viewer/ResourceViewerBlock.vue:27]: fix ESLint issues

File: src/components/interaction-designer/resource-viewer/ResourceViewerBlock.vue [hunk 3]
```diff
@@ -34,18 +44,21 @@ import {PropType} from 'vue'
 
 export default {
   name: 'ResourceViewerBlock',
+  mixins: [lang],
   props: {
-    block: Object as PropType<IBlock>,
+    block: {
+      required: true,
+      type: Object as PropType<IBlock>,
+    },
```
Comment (bulatgab) [inline src/components/interaction-designer/resource-viewer/ResourceViewerBlock.vue:52]: fix ESLint issues

File: src/components/interaction-designer/resource-viewer/ResourceViewerBlock.vue [hunk 4]
```diff
@@ -34,18 +44,21 @@ import {PropType} from 'vue'
...
-  mixins: [lang],
   computed: {
     ...mapGetters('flow', ['block_classesConfig']),
     ...mapGetters('builder', ['isEditable']),
     blockLabel() {
       return this.block.label > '' ? this.block.label : lang.trans('flow-builder.untitled-block')
     },
     hasContent() {
-      return this.block_classesConfig[this.block.type].is_interactive === true
+      return this.block.config.prompt !== undefined
```
Comment (bulatgab) [inline src/components/interaction-designer/resource-viewer/ResourceViewerBlock.vue:61]: main change

Comment (bulatgab) [reply src/components/interaction-designer/resource-viewer/ResourceViewerBlock.vue:61]: @object-required 
You used another criteria in voto5 for this:
```
...
    const promptValidationStatus = rootState.validation.validationStatuses[`resource/${block?.config?.prompt}`]?.isValid

    return {
      isContentBlock: promptValidationStatus !== undefined,
      ...
    }
...
```

What would you suggest we could do to make them consistent? Keep two different ones / use my code / use your code?

File: src/components/interaction-designer/resource-viewer/ResourceViewerBlock.vue [hunk 5]
```diff
@@ -34,18 +44,21 @@ import {PropType} from 'vue'
...
   },
-  mixins: [lang],
   computed: {
     ...mapGetters('flow', ['block_classesConfig']),
     ...mapGetters('builder', ['isEditable']),
     blockLabel() {
       return this.block.label > '' ? this.block.label : lang.trans('flow-builder.untitled-block')
     },
     hasContent() {
-      return this.block_classesConfig[this.block.type].is_interactive === true
```
Comment (safydy-r) [inline src/components/interaction-designer/resource-viewer/ResourceViewerBlock.vue:48]: good spot

**Review by safydy-r — APPROVED** [review]
No concern from me, I'm good. But let's hear what @object-required says about your question. Approving though

File: src/components/interaction-designer/resource-viewer/ResourceViewerBlock.vue (hunk 4 above)
Comment (object-required) [reply src/components/interaction-designer/resource-viewer/ResourceViewerBlock.vue:61]: I believe `this.block.config.prompt !== undefined` is more straightforward, and it doesn't make us rely on the properly working validation. Let's go with yours :-)

**Review by object-required — APPROVED** [review]

**Pull request merged**